design/one-pager-merge-jets-proposal.md

```diff
@@ -0,0 +1,370 @@
+the community that makes them mature enough to be used in production. While Jet
+providers are working on top of the mature TF providers that have been
+maintained longer than Crossplane's existence in some cases, they will still
+need contributions from the community to get to a certain level of maturity. If
+there is a single provider, then the community wouldn't have to deal with two
+implementations of the same API, trying to cover ground on both. The bugs would
+be fixed once and for everyone.
+
+Another point is that, just like users, contributors would also face a decision
+they may have a hard time to make - invest in classic one which will see
+decreased usage or Jet one which will be deprecated at one point, even if years
+away. If there is a single mixed provider, then everyone would know the path
+forward and be assured that the next step for that provider is only additive
+towards the end goals; not deprecation or removal of their contributions in
+favor of a temporary approach.
+
+### Cross-resource References
+
+Crossplane has cross-resource references just like Kubernetes; you get to
+specify the name of the referenced object or labels to select it and the
+controller of the referencer takes care of the rest, which includes the type and
+what field to look for to get the information.
+
+This is possible only if there is a single type to target and if the controller
+knows which field should be used as the source of information. Take a look at
+the following example:
+
+```yaml
+# User can give this information manually but if vpcIdRef is used, controller knows
+# which field of VPC object to copy here.
+vpcId: vpc-2314334
+# User can give only the name of the VPC object and then vpcId will be populated.
+vpcIdRef:
+  name: my-vpc
+# User can give only a selector and then the vpcIdRef will be populated.
+vpcIdSelector:
+  matchLabels:
+    app: ola
+```
+
+When we have Jet and classic providers, one does not know about the other's
+types. Hence, the only way to do references is to give the type of the
+referenced objct and the field path of the information on that type, something
+roughly like the following:
+
+```yaml
+vpcId: vpc-2314334
+vpcIdRef:
+  apiVersion: ec2.aws.crossplane.io/v1beta1
+  kind: VPC
+  fieldPath: metadata.annotations[crossplane.io/external-name]
+vpcIdSelector:
+  apiVersion: ec2.aws.crossplane.io/v1beta1
+  kind: VPC
+  fieldPath: metadata.annotations[crossplane.io/external-name]
+  matchLabels:
+    app: ola
+```
+
+While this referencing style is useful for many scenarios and being discussed
+[here][generic-references], it's not the same as how Kubernetes resources
+reference each other and requires more upfront information. We could possibly
+make it work like native with a default and let users optionally give different
+`kind`, `apiVersion` and `fieldPath`, there are other concerns with the approach
+listed in its discussion.
```

Comment:
I have a hunch that we'll want generic references eventually, and I'd prefer for us to align on one way to handle references (i.e. deprecate our current way of doing cross resource refs once we figure out something better). That said, I do appreciate that what this document proposes allows us to kick that can down the road a bit instead of blocking on solving it now.